src/System.Text.Json/tests/Serialization/Object.WriteTests.cs

```diff
@@ -53,5 +53,27 @@ public class ObjectObject
         {
             public object Object { get; set; }
         }
+
+        [Fact]
+        public static void WriteObject_PublicIndexer()
+        {
+            var indexer = new Indexer();
+            indexer[42] = 42;
+            indexer.NonIndexerProp = "Value";
+            Assert.Equal(@"{""NonIndexerProp"":""Value""}", JsonSerializer.ToString(indexer));
```

Comment:
nit: 

We are not consistent here across the test set, but we should generally try to test all the public overloads (which includes `ToUtf8Bytes`, and the `WriteAsync` API). The same applies on the read side. Though, I wouldn't single this PR/change out for that. This is something we need to do all up, so just pointing it out on new tests we are adding.

cc @layomia, @buyaa-n, @JeremyKuhne